lib/common/transforms.js

```diff
@@ -696,6 +696,40 @@ module.exports = {
     }
   },
 
+  /**
+   * Scales the pixel value to rem, and adds 'rem' to the end of non-zero values. If you define a "basePxFontSize" on the platform in your config, it will be used to scale the pixel value, otherwise 16 (default web font size) will be used.
+   *
+   * ```js
+   * // Matches: prop.attributes.category === 'size'
+   * // Returns:
+   * "0"
+   * "1rem"
+   * ```
+   */
+  'size/pxToRem': {
+    type: 'value',
+    matcher: (prop) => prop.attributes.category === 'size',
+    transformer: (prop, options) => {
+      const baseFont = (options && options.basePxFontSize) || 16;
+      const floatVal = parseFloat(prop.value);
+
+      if (isNaN(floatVal)) {
+        throwSizeError(prop.name, prop.value, 'rem');
+      }
+
+      if (floatVal === 0) {
+        return '0';
+      }
+
+      if (String(prop.value).indexOf('px') === -1) {
```

Comment:
I like the spirit of this check, but no other transforms right now check for a unit, making this an outlier. What are your thoughts on removing this check in this transform, and then creating some more generic size transforms that look for units and apply the proper transform? I think it would be good to have these more intelligent transforms for people that want to specify borders in px but everything else in rem.

Reply:
Yeah, I can drop the check. Using `parseFloat` allows safely ignoring the unit if it is present.

I'm not entirely sure what you mean by "[create] some more generic size transforms that look for units and apply the proper transform"? Do you mean something like `font/remToPx`, where `matcher: isFontSize`? That's the only example that I can think of.

Reply:
I'm thinking of a transform that would transform web-based values (in either px or rem) and covert them to dp/sp on Android and pts (CGFloats) on iOS. And it would do a check like above to tell whether the number should be directly translated (px are roughly equivalent to dp/sp/pt) or multiplied by the basePxFontSize for rem units. That way someone could define sizes in pixels and rems and not need 2 separate transforms. Does that make sense? Maybe I just need to write a PR with some example code...

Reply:
Something like this?

name|matcher|notes
--|--|--
`size/webToSp`|`isFontSize()`|if "px", replace "px" with "sp"<br>if "rem", scale and append "sp"
`size/webToDp`|`isNotFontSize()`|if "px", replace "px" with "dp"<br>if "rem", scale and append "dp"
`size/webToPt`|`isSize()`|if "px", replace "px" with "pt"<br>if "rem", scale and append "pt"

The one thing I can't wrap my head around is from your original comment...
> I think it would be good to have these more intelligent transforms for people that want to specify borders in px but everything else in rem.

Looking at existing transforms: 
* `*ToSp` always uses `isFontSize`, so it is already specific to fonts.
* `*ToDp` always uses `isNotFontSize`, so, I think, it only applies to borders, spacing, etc.
* `*ToPt` aways uses `isSize`, so it applies to everything.

Android appears to already handle making a distinction between fonts and borders. How would the same be done to ios, or the web for that matter?

Apologies, I have zero experience with andoid and ios development, which I think is causing me all kinds of confusion.

Sidenote, I dropped the "px" check.

Reply:
Your table is exactly what I was imagining. Right now the existing transforms assume all sizes are either defined in rem or px, but could never be a combination of them. For example I could not define my borders in px and my font size and paddings in rem because the transforms assume all sizes are one or the other (using size/remToSp and size/remToDp or size/pxToSp and size/pxToDp). With the current transforms I could have fonts in rems, and paddings and borders in px, but I couldn't have font and padding in rem and borders in pixels. Hopefully that makes sense 🥴

Reply:
Hmm, I think I'm getting confused because I'm trying to understand how the `size/webTo*` transforms apply to my scenario, when I think they might not actually apply.

In my scenario, the design system has everything in pixels. Our only target for now is the web (no mobile) and all values are to be converted to rems. Therefore, I want to use something like `size/pxToRem` to avoid manually performing the px-to-rem conversions.

If I'm understanding your comments, another scenario is a design system that defines mixed units. In that case, the expectation would be that any rem values would have been manually converted prior to being dropped into a property.json file? And, therefore, this scenario would never use `size/pxToRem` at the same time? Is that correct?

Reply:
I think what I'm ultimately trying to get at... is adding the `size/webTo*` transforms appropriate for this PR, or should they be the star of their own PR?

Reply:
I think we should address it in a separate PR, I think this one is good enough as it is!